### PR TITLE
feat(flow): query table schema&refactor

### PR DESCRIPTION
**Pull request by discord9**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

added `TableSource` to get table name, id and schema from metasrv, and other minor refactor

Please explain IN DETAIL what the changes are in this PR and why they are needed:

- added `TableSource` to help flow node get table name, id and schema from metasrv
- refactor `SourceSender` to allow unbounded broadcast

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

### CODE REVIEW

File: src/flow/src/adapter/node_context.rs [hunk 1]
```diff
@@ -74,38 +60,94 @@ pub struct FlownodeContext {
...
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl SourceSender {
+    pub fn new() -> Self {
+        Self {
+            sender: broadcast::Sender::new(BROADCAST_CAP),
+            send_buf: Default::default(),
```
Comment (zhongzc) [inline src/flow/src/adapter/node_context.rs:82]: Move to `default` and the `new` can be removed

**Review by zhongzc — APPROVED** [review]
Almost LGTM, very readable PR! 😍

File: src/flow/src/adapter/table_source.rs [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+            .get(*table_id)
+            .await
+            .map_err(BoxedError::new)
+            .context(ExternalSnafu)?
+            .with_context(|| UnexpectedSnafu {
+                reason: format!("Table id = {:?}, couldn't found table name", table_id),
+            })
+            .map(|name| name.table_name())
+            .map(|name| [name.catalog_name, name.schema_name, name.table_name])
+    }
```
Comment (fengjiachun) [inline src/flow/src/adapter/table_source.rs:90]:
```suggestion
    }

```

**Review by fengjiachun — APPROVED** [review]

**Review by waynexia — APPROVED** [review]
Others LGTM

File: src/flow/src/adapter.rs [hunk 3]
```diff
@@ -17,9 +17,16 @@
...
-pub(crate) use node_context::{FlowId, FlownodeContext, TableName};
+pub(crate) use node_context::FlownodeContext;
+pub(crate) use table_source::TableSource;
 
 mod worker;
 
 pub const PER_REQ_MAX_ROW_CNT: usize = 8192;
+// TODO: refactor common types for flow to a separate module
+/// FlowId is a unique identifier for a flow task
+pub type FlowId = u64;
```
[the remark this answers is not in the document]
Comment (discord9) [reply src/flow/src/adapter.rs]: > Is this the same `FlowId` in proto? That one is `u32` https://github.com/GreptimeTeam/greptime-proto/blob/a11db14b8502f55ca5348917fd18e6fcf140f55e/proto/greptime/v1/ddl.proto#L161

changed to u32 now

**Pull request merged**